memory-viz/src/automate.ts

```diff
@@ -149,7 +149,12 @@ function drawAutomatedOtherItems(
     const START_X = sf_endpoint + PADDING;
 
     for (const item of objs) {
-        if (item.name !== "BLANK") {
```

Comment:
This change seems to be overcomplicating things. The change you've made now requires that the `type` attribute be `"BLANK"`. `name` shouldn't be checked at all. It seems you should just change `item.name` to `item.type`, here and below.